src/coreclr/src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs

```diff
@@ -326,6 +326,16 @@ public static string GetTypeInfoName(ITypeInfo typeInfo)
         /// </summary>
         public static IntPtr /* IUnknown* */ GetIUnknownForObject(object o)
         {
+            if (o is null)
+            {
+                throw new ArgumentNullException(nameof(o));
+            }
+
+            // Passing null as the ComWrapper implementation will use the globally registered wrappper (if available)
+            IntPtr ptrMaybe = ComWrappers.GetOrCreateComInterfaceForObjectInternal(impl: null, o, CreateComInterfaceFlags.None);
```

Comment:
>None [](start = 122, length = 4)

I think we need to pass the `TrackerSupport` in order to match the semantics of the built-in support.